play-services-core/src/main/res/xml/preferences_start.xml

```diff
@@ -65,9 +65,15 @@
             android:title="@string/service_name_location"/>
     </PreferenceCategory>
     <PreferenceCategory android:layout="@layout/preference_category_no_label" android:key="prefcat_footer">
+        <SwitchPreferenceCompat
+            android:defaultValue="false"
+            android:icon="@drawable/ic_hide_app_icon"
+            android:key="hideAppIcon"
+            android:summary="@string/pref_hide_app_icon_summary"
+            android:title="@string/pref_hide_app_icon_title" />
```

Comment:
Preferences should not be directly modified from the `:ui` process as shared preferences are not safe for multi-process use. That's why we generally use the `SettingsProvider` to store settings.

However, in this case, the settings provider is not needed, as the ground truth of the component enabled state is in the PackageManager. With your code, if someone enables or disables the component without using this setting (e.g. by directly invoking `pm disable` via `adb`), the setting would no longer match the actual live state.

I suggest you make the setting non-persistent (by settings `android:persistent="false"`) and configure it from code by taking the value from `getComponentEnabledSetting` when realizing the setting.